pkg/pingpong/pingpong.go

```diff
@@ -69,7 +69,11 @@ func (s *Service) Ping(ctx context.Context, address swarm.Address, msgs ...strin
 		return 0, fmt.Errorf("new stream: %w", err)
 	}
 	defer func() {
-		go stream.FullClose()
+		if err != nil {
+			_ = stream.Reset()
```

Comment:
Here again, we only do read and write operations, so if they errored, the stream is in an errored state and the reset is ineffectual